src/iterators/not_iterator.c

```diff
@@ -226,6 +226,74 @@ static IteratorStatus NI_SkipTo_Optimized(QueryIterator *base, t_docId docId) {
   return rc;
 }
 
+// Revalidate for NOT iterator - Non-optimized version.
+static ValidateStatus NI_Revalidate_NotOptimized(QueryIterator *base) {
+  NotIterator *ni = (NotIterator *)base;
+
+  // 1. Revalidate the child iterator
+  ValidateStatus child_status = ni->child->Revalidate(ni->child);
```

Comment:
so if VALIDATE_MOVED from child we do not care?

Reply:
With the Not iterator, the child should always be ahead of the internal counter/wildcard, as it has to mark the next invalid result. The only exceptions are initially when they are both on id 0, and when the child is at EOF (and then the counter/wildcard will get ahead of the child).
If we got `VALIDATE_MOVED`, it means the child is **at a new valid result**, so it's not any of the exceptions. If it was ahead/at the same ID as the interlan wildcard, it's even "more ahead" now. This should not affect the current result